shared/actions/teams/index.js

```diff
@@ -606,43 +609,74 @@ function* _createChannel(action: Constants.CreateChannel) {
-  yield Saga.put(replaceEntity(['teams', 'teamNameToLoading'], I.Map([[teamname, true]])))
-  yield Saga.call(RPCTypes.teamsSetTeamMemberShowcaseRpcPromise, {
-    param: {
-      isShowcased: enabled,
-      name: teamname,
-    },
+const _setPublicity = function({payload: {teamname, settings}}: Constants.SetPublicity, state: TypedState) {
+  const waitingKey = {key: `setPublicity:${teamname}`}
+  const teamSettings = state.entities.getIn(['teams', 'teamNameToTeamSettings', teamname], {
+    open: false,
+    joinAs: RPCTypes.teamsTeamRole['reader'],
   })
-  // getDetails will unset loading and update the store with the new value
-  yield Saga.put((dispatch: Dispatch) => dispatch(Creators.getDetails(teamname)))
-}
-
-function* _setPublicityTeam(action: Constants.SetPublicityTeam) {
-  const {payload: {enabled, teamname}} = action
-  yield Saga.put(replaceEntity(['teams', 'teamNameToLoading'], I.Map([[teamname, true]])))
-  yield Saga.call(RPCTypes.teamsSetTeamShowcaseRpcPromise, {
-    param: {
-      isShowcased: enabled,
-      name: teamname,
-    },
+  const teamPublicitySettings = state.entities.getIn(['teams', 'teamNameToPublicitySettings', teamname], {
+    anyMemberShowcase: false,
+    member: false,
+    team: false,
   })
-  // getDetails will unset loading and update the store with the new value
-  yield Saga.put((dispatch: Dispatch) => dispatch(Creators.getDetails(teamname)))
+  const openTeam = teamSettings.open
+  const openTeamRole = teamSettings.joinAs
+  const publicityAnyMember = teamPublicitySettings.anyMemberShowcase
+  const publicityMember = teamPublicitySettings.member
+  const publicityTeam = teamPublicitySettings.team
+
+  const calls = []
+  if (openTeam !== settings.openTeam || (settings.openTeam && openTeamRole !== settings.openTeamRole)) {
+    calls.push(
+      Saga.callAndWrap(RPCTypes.teamsTeamSetSettingsRpcPromise, {
+        param: {
+          name: teamname,
+          settings: {
+            joinAs: RPCTypes.teamsTeamRole[settings.openTeamRole],
+            open: settings.openTeam,
+          },
+        },
+      })
+    )
+  }
+  if (publicityAnyMember !== settings.publicityAnyMember) {
+    calls.push(
+      Saga.callAndWrap(RPCTypes.teamsSetTeamShowcaseRpcPromise, {
+        param: {
+          anyMemberShowcase: settings.publicityAnyMember,
+          name: teamname,
+        },
+      })
+    )
+  }
+  if (publicityMember !== settings.publicityMember) {
+    calls.push(
+      Saga.callAndWrap(RPCTypes.teamsSetTeamMemberShowcaseRpcPromise, {
+        param: {
+          isShowcased: settings.publicityMember,
+          name: teamname,
+        },
+      })
+    )
+  }
+  if (publicityTeam !== settings.publicityTeam) {
+    calls.push(
+      Saga.callAndWrap(RPCTypes.teamsSetTeamShowcaseRpcPromise, {
+        param: {
+          isShowcased: settings.publicityTeam,
+          name: teamname,
+        },
+      })
+    )
+  }
+  return Saga.all([
+    Saga.all(calls),
```

Comment:
I'm unfamiliar with `callAndWrap`, why not just use `Saga.call` for the elements in `calls` and get rid of `Saga.all` here?

Reply:
It's duplicating the setup @MarcoPolo used for the manage channels RPCs -- there's a new third arg to takeEveryPure that helps out with the increment/decrement and passing through errors from the RPCs.